src/components/map/FeatureLoadService.js

```diff
@@ -0,0 +1,64 @@
+
+goog.provide('ga_featureload_service');
+
+goog.require('ga_map_service');
+goog.require('ga_styles_service');
+
+(function() {
+  var module = angular.module('ga_featureload_service', [
+    'ga_map_service',
+    'ga_styles_service'
+  ]);
+
+  /*
+   *This service checks if features are loaded.
+   */
+
+  module.provider('gaFeatureLoadManager', function() {
```

Comment:
I don't think think you need to create a new service.
The GeoJSON sources are loaded in the gaLayers service, [here](https://github.com/geoadmin/mf-geoadmin3/blob/master/src/components/map/MapService.js#L874) using promise. So you can uses this promise instead of watching a geosjson layer to be added.

Fill a `geojsonPromises`  array in the `gaLayers` service and add also a `getLayerPromise(bodId)` function, which does the same as the `getLoadPromise`.

Reply:
We created the service because of decoupling. We don't want to overload gaLayers with yet another functionality.

Reply:
I can understand that but the promise already exists in the gaLayers so using it makes sense.

Reply:
Yes, that's true, but we'll need to manage the promises in gaLayers - the original reason for the service.

But I concur - let's go that route.

Reply:
As far as I understood, another reason for our approach was that we never call the function `setLayerSource()` because of the condition [here](https://github.com/geoadmin/mf-geoadmin3/blob/381c1db6046356d44d65fea180a80bb9186af402/src/components/map/MapService.js#L896) (pls correct me if I'm wrong!). @oterral: Do you have a suggestion how to "bypass" this `if`? Thx in advance!

Reply:
Yes but it shouldn't be the case. 
There is code duplication in MapService and in RealTimeLayersService. We should fix this in another PR. For now, remove this if and launch the timer instead of settig source [herer](https://github.com/geoadmin/mf-geoadmin3/blob/54657c28b10cb91118c60110c5105636d2ba26ea/src/components/map/RealtimeLayersService.js#L97)